crypto/keys/ecdsa/ecdsa_pubkey_test.go

```diff
@@ -0,0 +1,154 @@
+package ecdsa
+
+import (
+	"crypto/ecdsa"
+	"encoding/hex"
+	"math/big"
+	"testing"
+
+	proto "github.com/gogo/protobuf/proto"
+	gogotypes "github.com/gogo/protobuf/types"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+)
+
+func TestPKSuite(t *testing.T) {
+	suite.Run(t, new(PKSuite))
+}
+
+type CommonSuite struct {
+	suite.Suite
+	pk cryptotypes.PubKey
+	sk cryptotypes.PrivKey
+}
+
+func (suite *CommonSuite) SetupSuite() {
+	sk, err := GenSecp256r1()
+	suite.Require().NoError(err)
+	suite.sk = sk
+	suite.pk = sk.PubKey()
+}
+
+type PKSuite struct{ CommonSuite }
```

Comment:
Why? Why we need CommonSuite and PKSuite?

Reply:
CommonSuite doesn't have any tests - it only contains common code (used both in PKSuite and SKSuite).

Reply:
Ahh ok makes sense :+1: but why are they public?

Reply:
It's not public. You can't import from `_test.go` files.